### PR TITLE
[doc] add some go doc in conn.go

### PR DESCRIPTION
**Pull request by linsite**
The  pull comes as the help of the issue #4 . If you guys see it O.K., I'll continue to add other missing comments. FYI, I'm happy to participant in an Open Source project. Thanks for your time.

### CODE REVIEW

File: conn.go [hunk 1]
```diff
@@ -54,6 +54,7 @@ type watchPathType struct {
 	wType watchType
 }
 
+// Dailer is the connection dialer function for ZK.
```
Comment (nemith) [inline conn.go:57]: Spelling mistake.  Should be `Dialer` not `Dailer`

File: conn.go [hunk 2]
```diff
@@ -66,6 +67,8 @@ type authCreds struct {
 	auth   []byte
 }
 
+// Conn is the connection object for a ZK client, it holds all the stuff one
+// needs to communicate with ZK server, all the properties should be private.
```
Comment (nemith) [inline conn.go:71]: I would simplify this a bit.   Something like

```
// Conn is the client connection and tracks all details for communication with the server.
```

File: conn.go (hunk 1 above)
Comment (nemith) [reply conn.go:57]: This is sorta redundant (i.e Dialer is a dialer).

Perhaps something like:

```
// Dialer is a function to be used to establish a connection to a single host.
```

**Review by nemith — CHANGES_REQUESTED** [review]
Thanks for doing this.  Good documentation or some documentation makes godocs easier to read so this is great and has been a long standing issue!

I have a couple of nits, but other than that it looks good.  If you could address them then we'll land this!

File: conn.go (hunk 2 above)
Comment (linsite) [reply conn.go]: Point taken. Thanks, I'll learn to improve in the future. :)

File: conn.go (hunk 1 above)
Comment (linsite) [reply conn.go]: Yeah, that's better, I'v modified it.

File: conn.go [hunk 3]
```diff
@@ -138,6 +141,8 @@ type response struct {
 	err  error
 }
 
+// Event is an Znode event sent by the ZK server.
```
[the remark this answers is not in the document]
Comment (linsite) [reply conn.go]: Motified as advices. :)

**Comment by nemith** [comment]
Looks great.  Thank you!

**Pull request merged**